### PR TITLE
handle new profile

### LINKED ISSUE #2: handle new profile
The EAR spec, as well as the go implementation, have moved to the 2023 profile.
We don't want to be left behind :-)

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #2

### CODE REVIEW

**Comment by thomas-fossati** [comment]
cc @paulhowardarm

File: src/ear.c [hunk 1]
```diff
@@ -100,31 +102,82 @@ int ear_jwt_verify(const char *ear_jwt, const uint8_t *pkey, size_t pkey_sz,
...
-    }
+  const char *status_s = json_string_value(status);
+
+  if (tier_from_string(status_s, ptier) == -1) {
+    (void)snprintf(err_msg, EAR_ERR_SZ, "unknown status \"%s\"", status_s);
+    goto err;
   }
 
-  (void)snprintf(err_msg, EAR_ERR_SZ, "unknown status \"%s\"", status);
+  json_decref(submods);
```
Comment (thomas-fossati) [inline src/ear.c]:
```suggestion
  json_decref(submods), submods = NULL;
```

File: src/ear.c [hunk 2]
```diff
@@ -100,31 +102,82 @@ int ear_jwt_verify(const char *ear_jwt, const uint8_t *pkey, size_t pkey_sz,
...
   }
 
-  for (unsigned i = 0; i < sizeof tiers / sizeof(char *); i++) {
-    if (!strcmp(status, tiers[i])) {
-      return status;
-    }
+  const char *status_s = json_string_value(status);
+
+  if (tier_from_string(status_s, ptier) == -1) {
+    (void)snprintf(err_msg, EAR_ERR_SZ, "unknown status \"%s\"", status_s);
```
Comment (thomas-fossati) [inline src/ear.c]:
```suggestion
    (void)snprintf(e, sizeof e, "unknown status \"%s\"", status_s);
```

**Review by setrofim — APPROVED** [review]
LGMT!

**Pull request merged**